src/nacl/bindings/crypto_shorthash.py

```diff
@@ -0,0 +1,37 @@
+# Copyright 2016 Donald Stufft and individual contributors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import absolute_import, division, print_function
+
+from nacl._sodium import ffi, lib
+
+BYTES = lib.crypto_shorthash_siphash24_bytes()
+KEYBYTES = lib.crypto_shorthash_siphash24_keybytes()
+
+
+def crypto_shorthash_siphash24(data, key):
+    """Compute a fast, cryptographic quality, keyed hash of the input data
+
+    :param data:
+    :type data: bytes
+    :param key: len(key) must be equal to
+                :py:data:`.KEYBYTES` (16)
+    :type key: bytes
+    """
+    assert len(key) == KEYBYTES
```

Comment:
Let's make this `if len(key) != KEYBTES: raise ValueError("Key length must be {0} bytes".format(KEYBYTES))` or something like that. Then we can update the test to look for ValueError